.github/workflows/build.yml

```diff
@@ -31,9 +31,66 @@ jobs:
           pattern: '*.sh'
           exclude: './.git/*,./vendor/*'
 
+  unit-tests:
+    name: Test
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Set up Go 1.14
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.14.7
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: verify license
+      run: make license-check
+
+    - name: verify dependencies
+      run: make deps
+
+    - name: verify tests
+      run: make test
+
+  e2e-tests:
+    needs: ['unit-tests']
```

Comment:
@shubham14bajpai Do we need e2e to be dependent on unit-tests. ? 

IMHO, `lint`, `unit-tests` and `e2e-tests` can be run in parallel. Only the final build/push needs to be dependent on all these 3 jobs.

Reply:
I think checking the modules using make deps and running unit tests to check for syntactical errors is good before building and running the images.